server/routes/utils.js

```diff
@@ -58,6 +59,30 @@ async function getTeamDocument(req, res, next) {
     return next()
 }
 
+// this middleware checks if the given teamIds are actual documents, and if so, populate them
+// it will also check if the given number of rounds fit can fit within a season
+async function _validateFixture(req, res, next) {
+    const { dateStart, dateFinish } = req.season
+    // Check valid teams
+    if (!(await allValidDocumentIds(req.body.teamIds, Team))) {
+        return res.status(404).json({ success: false, error: 'Some team does not exist' })
+    }
+    // Check team is added to grade
+    req.body.teamIds.forEach(team => {
+        if (!req.grade.teams.includes(team)) {
+            console.log(team)
+            return res.status(400).json({ success: false, error: 'Team is not added to grade' })
+        }
+    })
```

Comment:
because the `return` is inside the function here even if it is hit, the rest of the code will still run (which could potentially result in an error, or even worse, continue as normal and modify the db)

```suggestion
    const notAdded = req.body.teamIds.some((team) => !req.grade.teams.includes(team)) 
    if (notAdded) {
        return res.status(400).json({ success: false, error: 'Team is not added to grade' })
    }
```

Reply:
o sht i actually didnt know return doesn't break out of a forEach, thanks for picking this up!